CONTRIBUTING.md

```diff
@@ -0,0 +1,169 @@
+
+If you don't feel ready to make a code contribution yet, no problem! You can also check out the [documentation issues](https://github.com/CopilotKit/CopilotKit/issues?q=is%3Aopen+is%3Aissue+label%3Adocumentation).
+
+# How do I make a code contribution?
+
+## Good first issues
+
+Are you new to open source contribution? Wondering how contributions work in our project? Here's a quick rundown.
+
+Find an issue that you're interested in addressing, or a feature that you'd like to add.
+You can use [this view](https://github.com/CopilotKit/CopilotKit/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) which helps new contributors find easy gateways into our project.
+
+## Step 1: Make a fork
+
+Fork the CopilotKit repository to your GitHub organization. This means that you'll have a copy of the repository under _your-GitHub-username/repository-name_.
+
+![Group 3](https://github.com/user-attachments/assets/7c2b8d15-87cf-4cc7-be86-5fadaebfad0b)
+
+## Step 2: Clone the repository to your local machine
+
+```
+git clone https://github.com/<your-GitHub-username>/CopilotKit
+
+```
+
+![Group 4](https://github.com/user-attachments/assets/e3e78b2b-eead-463b-858b-8d40e4cb18e9)
+
+## Step 3: Prepare the development environment
+
+### 1)Install Prerequisites
+- Node.js 20.x or later
+- pnpm v9.x installed globally (npm i -g pnpm@^9)
+- Turborepo v2.x installed globally (npm i -g turbo@2)
+
+### 2)Install Dependencies
+To install the dependencies using pnpm
+Go inside project folder and run :
+
+```jsx
+pnpm install
+```
+### 3)Build Packages
+To make sure everything works, let’s build all packages once:
+
+```jsx
+turbo run build
+```
+
+## Step 4: Create a branch
+
+Create a new branch for your changes.
+In order to keep branch names uniform and easy-to-understand, please use the following conventions for branch naming.
+Generally speaking, it is a good idea to add a group/type prefix to a branch.
+Here is a list of good examples:
+
+- for docs change : docs/<ISSUE_NUMBER>-<CUSTOM_NAME>
+- for new features : feat/<ISSUE_NUMBER>-<CUSTOM_NAME>
+- for bug fixes : fix/<ISSUE_NUMBER>-<CUSTOM_NAME>
+
+```jsx
+git checkout -b <new-branch-name-here>
+```
+
+## Step 5: Make your changes
+
+Now that everything is set up and works as expected, you can get start developing or update the code with your bug fix or new feature.
+
+```jsx
+# To start all packages in development mode
+turbo run dev
+ 
+# Start a specific package in development mode
+turbo run dev --filter="@copilotkit/package-name"
+```
+
+## Step 6: Add the changes that are ready to be committed
+
+Stage the changes that are ready to be committed:
+
+```jsx
+git add .
+```
+
+## Step 7: Commit the changes (Git)
+
+Commit the changes with a short message. (See below for more details on how we structure our commit messages)
+
+```jsx
+git commit -m "<type>(<package>): <subject>"
+```
+
+## Step 8: Push the changes to the remote repository
```

Comment:
I think you should also mention pulling the latest changes from remote before pushing to avoid conflicts